resilience4j-framework-common/src/test/java/io/github/resilience4j/common/bulkhead/configuration/ThreadPoolBulkheadConfigurationPropertiesTest.java

```diff
@@ -0,0 +1,151 @@
+        sharedProperties.setCoreThreadPoolSize(2);
+        sharedProperties.setQueueCapacity(2);
+
+        ThreadPoolBulkheadConfigurationProperties.InstanceProperties backendWithDefaultConfig = new ThreadPoolBulkheadConfigurationProperties.InstanceProperties();
+        backendWithDefaultConfig.setBaseConfig("default");
+        backendWithDefaultConfig.setCoreThreadPoolSize(3);
+
+        ThreadPoolBulkheadConfigurationProperties.InstanceProperties backendWithSharedConfig = new ThreadPoolBulkheadConfigurationProperties.InstanceProperties();
+        backendWithSharedConfig.setBaseConfig("sharedConfig");
+        backendWithSharedConfig.setCoreThreadPoolSize(4);
+
+        ThreadPoolBulkheadConfigurationProperties bulkheadConfigurationProperties = new ThreadPoolBulkheadConfigurationProperties();
+        bulkheadConfigurationProperties.getConfigs().put("default", defaultProperties);
+        bulkheadConfigurationProperties.getConfigs().put("sharedConfig", sharedProperties);
+
+        bulkheadConfigurationProperties.getBackends()
+            .put("backendWithDefaultConfig", backendWithDefaultConfig);
+        bulkheadConfigurationProperties.getBackends()
+            .put("backendWithSharedConfig", backendWithSharedConfig);
+
+        //When
+        //Then
+        try {
+            assertThat(bulkheadConfigurationProperties.getBackends().size()).isEqualTo(2);
+            // Should get default config and core number
+            ThreadPoolBulkheadConfig bulkhead1 = bulkheadConfigurationProperties
+                .createThreadPoolBulkheadConfig("backendWithDefaultConfig",
+                    compositeThreadPoolBulkheadCustomizer());
+            assertThat(bulkhead1).isNotNull();
+            assertThat(bulkhead1.getCoreThreadPoolSize()).isEqualTo(3);
+            assertThat(bulkhead1.getQueueCapacity()).isEqualTo(1);
+            // Should get shared config and overwrite core number
+            ThreadPoolBulkheadConfig bulkhead2 = bulkheadConfigurationProperties
+                .createThreadPoolBulkheadConfig("backendWithSharedConfig",
+                    compositeThreadPoolBulkheadCustomizer());
+            assertThat(bulkhead2).isNotNull();
+            assertThat(bulkhead2.getCoreThreadPoolSize()).isEqualTo(4);
+            assertThat(bulkhead2.getQueueCapacity()).isEqualTo(2);
+            // Unknown backend should get default config of Registry
+            ThreadPoolBulkheadConfig bulkhead3 = bulkheadConfigurationProperties
+                .createThreadPoolBulkheadConfig("unknownBackend",
+                    compositeThreadPoolBulkheadCustomizer());
+            assertThat(bulkhead3).isNotNull();
+            assertThat(bulkhead3.getCoreThreadPoolSize())
+                .isEqualTo(ThreadPoolBulkheadConfig.DEFAULT_CORE_THREAD_POOL_SIZE);
+        } catch (Exception e) {
+            System.out.println(
```

Comment:
I think that log instead of System.out.println would be more elegant.

Reply:
I removed try catch. 
Thank you.